.github/workflows/publish-otel-php-base-docker-image.yml

```diff
@@ -15,16 +18,22 @@ jobs:
       - name: check out the repo
         uses: actions/checkout@v2
 
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
       - name: Login to GitHub Container Registry
         uses: docker/login-action@v1
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Push to ghcr.io
+      - name: Build and push ${{ matrix.php-version }} to ghcr.io
         uses: docker/build-push-action@v2
         with:
           push: true
           file: docker/Dockerfile
-          tags: ghcr.io/open-telemetry/opentelemetry-php/opentelemetry-php-base:latest
+          build-args: PHP_VERSION=${{ matrix.php-version }}
+          tags: ghcr.io/open-telemetry/opentelemetry-php/opentelemetry-php-base:${{ matrix.php-version }}
+          cache-from: type=gha,scope=${{ matrix.php-version }}
```

Comment:
Using github action cache - this speeds up builds substantially (grpc!) but as a side-effect we might not get updated dependencies. Is it worth the trade-off?

Reply:
It Looks like there is no way to configure a TTL or invalidate the cache. Maybe add some extra key to the cache-key to allow for invalidating when needed?

Reply:
The easiest way I can see is to add an extra variable to the key, that we can change (eg CACHE_VERSION) - but, that just sounds like troubleshooting and extra work for maintainers.
Given that the build takes around 20 minutes, and only runs against `main`, I think we will have less surprises in the future without cache.

Reply:
Yap, I agree. Actually the image build doesn't even need to be run after each merge, since it's not using the PHP code anyway, so once a week should be enough, no?

Reply:
Ooh, that's a good idea. It could run to schedule, and we have the option of manually running it as required.